src/widgets/connector.ts

```diff
@@ -378,17 +378,45 @@ export class ConnectorView extends DOMWidgetView {
 
         const connectionTemplate = this.connectionsTemplates[key];
 
+        const form: { [key: string]: any } = {};
+
+        // get previous selected connection
+        const previousSelect = sessionStorage.getItem("selectConnection");
+
+        if (previousSelect) {
+          const prevConnectionTemplate = this.connectionsTemplates[previousSelect];
+          const { fields } = prevConnectionTemplate;
+          fields.forEach((field: { id: string; default?: string }) => {
+            const id = field.id;
+            const defaultValue = field.hasOwnProperty("default")
+              ? field["default"]
+              : "";
+            const formField = <HTMLSelectElement>this.el.querySelector(`#${id}`);
+            if (formField && formField.value != defaultValue) {
+              form[id] = formField.value;
+            }
+          });
+        }
+
+        sessionStorage.setItem("fieldInputs", JSON.stringify(form));
```

Comment:
where is this sessionStorage coming from? I guess the variable was already declared because I don't see any new declaration

Reply:
It's part of the Web storage API: https://developer.mozilla.org/en-US/docs/Web/API/Web_Storage_API